arelle/ValidateDuplicateFacts.py

```diff
@@ -167,6 +172,105 @@ def areWithinRoundingError(self) -> bool:
                 return False
         return True
 
+    def deduplicateCompleteSubsets(self) -> list[ModelFact]:
+        """
+        :return: A list of the first fact found for each unique decimals/value combination.
+        """
+        seenKeys = set()
+        results = []
+        for fact in self.facts:
+            key = (fact.decimals, fact.xValue)
+            if key in seenKeys:
+                continue
+            seenKeys.add(key)
+            results.append(fact)
+        return results
+
+    def deduplicateConsistentPairs(self) -> list[ModelFact]:
+        """
+        First performs deduplication of complete duplicates,
+        then removes from the remaining facts any fact that is consistent with a higher precision fact.
+        :return: A subset of the facts where the fact of lower precision in every consistent pair has been removed.
+        """
+        facts = self.deduplicateCompleteSubsets()
+        if not self.areNumeric:
+            # Consistency is equivalent to completeness for non-numeric facts
+            return facts
+        decimalsMap = defaultdict(list)
+        for fact in facts:
+            decimals = self.getDecimals(fact)
+            assert decimals is not None
+            decimalsMap[decimals].append(fact)
+        sortedDecimals = sorted(decimalsMap.keys())
+        results = set(facts)
+
+        for a, decimalsA in enumerate(sortedDecimals[:len(sortedDecimals)]):
```

Comment:
The intent was that this loop did not iterate on the final group (the group of highest precision) since facts in that group will never be removed due to having a higher-precision duplicate. I see now that I incorrectly expected the index to be exclusive.